src/downshift.js

```diff
@@ -786,7 +786,12 @@ class Downshift extends Component {
 
   inputHandleKeyDown = event => {
     const key = normalizeArrowKey(event)
-    if (key && this.inputKeyDownHandlers[key]) {
+    if (
```

Comment:
maybe we can have a better condition for reading. why don't we add an if to the top of the function, checking event.which and is composing and return from function if one of them is true? in that way we don't need to normalize any arrow key and we can have two clean conditions.

Also I think the `event.which !== 229` was useless there since `normalizeArrowKey` checks if key is 37-40.
```suggestion
    inputHandleKeyDown = event => {
       if (event.which === 229 || event.isComposing) { return }
       const key = normalizeArrowKey(event)
       // bla bla the rest
    }
```

What do you think? Also I tried to find out what is the exact issue with this. Probably I kind of got the idea, but since you are native to it and know the whole thing, can you explain in detail what happens here? That GIF you posted is helpful, true, but for me, without knowing that IME composition existed, is hard to figure out. :D

Reply:
also before the first IF we can add a comment with this PR and also a short description of the problem, like `IME composing issue`

Reply:
Yes, I agree with it is bit unreadable. Actually  but if I retuned something in the event handler, Jest warned the test coverage doesn't reached to 100%. that's why I wrote like that.

Reply:
> maybe we can have a better condition for reading.

I agree with it is a bit unreadable. Initially, I've implemented it as you said above. (febf5a4fad243d2a5e45e8012c1b4b67a03f87de)
But since I'm not familiar with RTL, I couldn't find the way to test what returned in the event handler. That's why I couldn't avoid it.

> Also I think the `event.which !== 229` was useless there since `normalizeArrowKey` checks if key is 37-40.

Oh, I didn't realize it. `KeyboardEvent.isComposing` was available in my browser so it worked correctly.

Anyway, If you know how to test the return value of the event handler could you teach me? It will also resolve the `normalizeArrowKey` issue.